operantanalysis/scripts/habit_training.py

```diff
@@ -0,0 +1,41 @@
+from operantanalysis import load_file, extract_info_from_file, reward_retrieval, lever_pressing
+import glob
+from tkinter import filedialog
+from tkinter import *  # noqa
+import pandas as pd
+import matplotlib
+matplotlib.use("TkAgg")
+from matplotlib import pyplot as plt  # noqa
+
+days = int(input("How many days would you like to analyze?"))
+column_list = ['Subject', 'Sex', 'Day', 'Training', 'Dippers', 'Dippers Retrieved', 'Retrieval Latency',
+               'Lever Presses']
+df = pd.DataFrame(columns=column_list)
+
+
+for i in range(days):
+    root = Tk()  # noqa
```

Comment:
you can overcome absence of DISPLAY on travis by using xvfb.  See e.g. how they do it in PsychoPy project, or how @DevXl is doing in his project: https://github.com/DevXl/double-drift/blob/mar10-beginning/.travis.yml#L28

Reply:
alternatively, if you do not hardcode matplotlib backend, you could instruct (e.g. via environment variable) for it to be set to e.g. agg, which would not require display to render, so you could test. E.g. that is what we do in pymvpa (`git grep -i matplotlib.*backend` in PyMVPA sources)